lib/cocoapods/installer/user_project_integrator/target_integrator.rb

```diff
@@ -255,6 +256,47 @@ def add_embed_frameworks_script_phase
           end
         end
 
+        # Updates all target script phases for the current target, including creating or updating, deleting
+        # and re-ordering.
+        #
+        # @return [void]
+        #
+        def update_target_script_phases
+          target_definition_script_phases = target.target_definition.script_phases
+          target_definition_script_phase_names = target_definition_script_phases.map { |k| k[:name] }
+          standard_script_phase_names = [CHECK_MANIFEST_PHASE_NAME, EMBED_FRAMEWORK_PHASE_NAME, COPY_PODS_RESOURCES_PHASE_NAME]
+          native_targets.each do |native_target|
+            # Delete script phases no longer present in the target definition.
+            native_target_script_phases = native_target.shell_script_build_phases.select { |bp| !bp.name.nil? && bp.name.start_with?(BUILD_PHASE_PREFIX) && !standard_script_phase_names.include?(bp.name.sub(BUILD_PHASE_PREFIX, '')) }
+            native_target_script_phases.each do |script_phase|
+              script_phase_name_without_prefix = script_phase.name.sub(BUILD_PHASE_PREFIX, '')
+              unless target_definition_script_phase_names.include?(script_phase_name_without_prefix)
+                native_target.build_phases.delete(script_phase)
+              end
+            end
+
+            # Create or update the ones that are expected to be.
+            target_definition_script_phases.each do |td_script_phase|
+              phase = TargetIntegrator.create_or_update_build_phase(native_target, td_script_phase[:name])
+              phase.shell_script = td_script_phase[:script]
+              phase.shell_path = td_script_phase[:shell_path] if td_script_phase.key?(:shell_path)
+              phase.input_paths = td_script_phase[:input_files] if td_script_phase.key?(:input_files)
+              phase.output_paths = td_script_phase[:output_files] if td_script_phase.key?(:output_files)
+              phase.show_env_vars_in_log = td_script_phase[:show_env_vars_in_log] ? '1' : '0' if td_script_phase.key?(:show_env_vars_in_log)
```

Comment:
honestly we should probably force this to be off. Having it enabled always results in a bad time for CI logs and similar.

Reply:
At this point it is up to the user whether they want it off or not, I think its off by default